src/S3Bucket/Components/FileUpload.vue

```diff
@@ -0,0 +1,140 @@
+      <VBtn
+        prepend-icon="mdi-file-upload"
+        v-bind="fileProps"
+        variant="flat"
+        color="primary"
+        size="x-large"
+        title="Upload New File"
+        class="px-4 flex-grow-1"
+        >Upload a File</VBtn
+      >
+    </template>
+    <VCard class="pa-3">
+      <VForm
+        @submit.prevent="fileUploadModal"
+        @update:model-value="(val) => (formIsValid = val)"
+      >
+        <VCardTitle class="w-100 d-inline-flex justify-space-between text-h5">
+          <div>
+            Upload file to
+            <span class="font-italic">{{
+              bucketPath ? bucketPath : 'Root folder'
+            }}</span>
+          </div>
+          <VBtn
+            icon="mdi-close"
+            title="Close"
+            data-dismiss="modal"
+            variant="text"
+            @click="onCancel"
+          />
+        </VCardTitle>
+        <VCardText>
+          <VFileInput
+            v-model="uploadFile"
+            :rules="requiredRule"
+            clearable
+            label="Upload File"
+          />
+        </VCardText>
+        <VCardActions class="d-flex justify-end px-3 mb-1">
+          <VTooltip location="start" :text="formError">
+            <template #activator="{ props: activatorProps }">
+              <VIcon
+                v-if="formError"
+                v-bind="activatorProps"
+                color="error"
+                size="large"
+                icon="mdi-alert-circle"
+                class="mt-1"
+              />
+            </template>
+          </VTooltip>
+          <VBtn
+            prepend-icon="mdi-close"
+            variant="flat"
+            size="large"
+            class="px-4 mx-2"
+            @click="onCancel"
+            >Cancel</VBtn
+          >
+          <VBtn
+            :loading="isUploading"
+            prepend-icon="mdi-file-upload"
+            :disabled="!formIsValid"
+            variant="flat"
+            color="primary"
+            size="large"
+            class="px-4"
+            type="submit"
+            >Upload to S3
+            <template #loader>Uploading...</template>
+          </VBtn>
+        </VCardActions>
+      </VForm>
+    </VCard>
+  </VDialog>
+</template>
+
+<script setup>
+import { inject, ref } from 'vue';
+import { convertObjectToMultiFormData } from '../../helpers/axiosHelper';
+import { useBuckets } from '../../helpers/useBuckets';
+
+const api = inject('api')
+const { bucketPath, bucketResource, refreshResource } = useBuckets(api)
+const emit = defineEmits(['update:closeDialog'])
+const isUploading = ref(false)
+const uploadFile = ref([])
+const uploadFileForm = ref({
+  bucket_name: bucketResource.value.name,
+  path: bucketPath.value
+})
+const formIsValid = ref(false)
+const formError = ref()
+const fileDialog = ref(false)
+const requiredRule = [(value) => value.length > 0 || 'This field is required']
+
+const onCancel = () => {
+  emit('update:closeDialog')
```

Comment:
The emit pattern `update:something` is meant to be used with a `something` prop to make v-model work on this component. That doesn't seem to be the intention here, so I'd suggest one of:
 - renaming this emit to just `closeDialog` or `cancel` or something.
 - I think you can simplify it out entirely. Since it looks like this event is just basically closing the VDialog in the parent (unless I'm missing something), VDialog can already do that with its v-model. Since VDialog is the child component here, it automatically binds its v-model to this component. You should be able to just use v-model in the parent (or use @update:modelValue in it) where you're using  `update_closeDialog` now.

Reply:
And same in FolderUpload